src/slurm_plugin/clustermgtd.py

```diff
@@ -707,8 +714,11 @@ def _find_unhealthy_slurm_nodes(self, slurm_nodes):
         unhealthy_static_nodes = []
         unhealthy_dynamic_nodes = []
         ice_compute_resources_and_nodes_map = {}
+        all_unhealthy_nodes = []
         for node in slurm_nodes:
             if not node.is_healthy(self._config.terminate_drain_nodes, self._config.terminate_down_nodes):
+                all_unhealthy_nodes.append(node)
```

Comment:
I think the idea is that the event publisher needs to publish different events for different types of unhealthy nodes and these may not divide up in the same way that clustermgtd needs to group them. So we decided just to hand off the list of all unhealthy nodes and let the event publisher do its own sorting. Of course in some cases this may lead to duplicated logic, but I figured it was better that way than to put any event specific logic into clustermgtd. What do you think?